recipes-bsp/u-boot/u-boot-distro-boot.bbappend

```diff
@@ -10,12 +10,161 @@ SRC_URI:append = " \
+CLOSE_CMD:mx8m-generic-bsp = "${IMX7_IMX8M_COMMON_CLOSE_CMD}"
+
+BANK_WORD:mx8qm-generic-bsp = "fuse_bank_word_1=0 722\
+\nfuse_bank_word_2=0 723\
+\nfuse_bank_word_3=0 724\
+\nfuse_bank_word_4=0 725\
+\nfuse_bank_word_5=0 726\
+\nfuse_bank_word_6=0 727\
+\nfuse_bank_word_7=0 728\
+\nfuse_bank_word_8=0 729\
+\nfuse_bank_word_9=0 730\
+\nfuse_bank_word_10=0 731\
+\nfuse_bank_word_11=0 732\
+\nfuse_bank_word_12=0 733\
+\nfuse_bank_word_13=0 734\
+\nfuse_bank_word_14=0 735\
+\nfuse_bank_word_15=0 736\
+\nfuse_bank_word_16=0 737\
+"
+FUSE_NUM:mx8qm-generic-bsp = "${FUSE_NUM_16}"
+STATUS_CMD:mx8qm-generic-bsp = "${AHAB_STATUS_CMD}"
+CLOSE_CMD:mx8qm-generic-bsp = "${AHAB_CLOSE_CMD}"
+
+BANK_WORD:mx8x-generic-bsp = "fuse_bank_word_1=0 730\
+\nfuse_bank_word_2=0 731\
+\nfuse_bank_word_3=0 732\
+\nfuse_bank_word_4=0 733\
+\nfuse_bank_word_5=0 734\
+\nfuse_bank_word_6=0 735\
+\nfuse_bank_word_7=0 736\
+\nfuse_bank_word_8=0 737\
+\nfuse_bank_word_9=0 738\
+\nfuse_bank_word_10=0 739\
+\nfuse_bank_word_11=0 740\
+\nfuse_bank_word_12=0 741\
+\nfuse_bank_word_13=0 742\
+\nfuse_bank_word_14=0 743\
+\nfuse_bank_word_15=0 744\
+\nfuse_bank_word_16=0 745\
+"
+FUSE_NUM:mx8x-generic-bsp = "${FUSE_NUM_16}"
+STATUS_CMD:mx8x-generic-bsp = "${AHAB_STATUS_CMD}"
+CLOSE_CMD:mx8x-generic-bsp = "${AHAB_CLOSE_CMD}"
+
+UENV_VARS_TO_DEL ?= ""
+# List of uEnv.txt variables not present during initial generation
+UENV_VARS_TO_DEL_EXTRA ?= "kernel_image \
+    ramdisk_image \
+    fdtdir \
+    bootargs \
+    kernel_image2 \
+    ramdisk_image2 \
+    fdtdir2 \
+    bootargs2 \
+    fdtfile2 \
+    kernel_image_type2 \
+    fuse_bank_word_1 \
+    fuse_bank_word_2 \
+    fuse_bank_word_3 \
+    fuse_bank_word_4 \
+    fuse_bank_word_5 \
+    fuse_bank_word_6 \
+    fuse_bank_word_7 \
+    fuse_bank_word_8 \
+    fuse_bank_word_9 \
+    fuse_bank_word_10 \
+    fuse_bank_word_11 \
+    fuse_bank_word_12 \
+    fuse_bank_word_13 \
+    fuse_bank_word_14 \
+    fuse_bank_word_15 \
+    fuse_bank_word_16 \
+    index \
+    temp \
+"
+
 do_compile:append () {
+    vars_to_del="${UENV_VARS_TO_DEL}"
+    if [ -z "${vars_to_del}" ]; then
+        vars_to_del="$(cat ${WORKDIR}/uEnv.txt.in | \
+                       sed -n -e 's/^\([A-Za-z_][A-Za-z0-9_]*\)=.*/\1/gp')"
+        # Needed to remove newlines from previous output
+        vars_to_del=$(echo ${vars_to_del})
+    fi
+    vars_to_del="${vars_to_del} ${UENV_VARS_TO_DEL_EXTRA}"
+
     bbdebug 1 "Building uEnv.txt..."
     sed -e 's/@@KERNEL_BOOTCMD@@/${KERNEL_BOOTCMD}/' \
         -e 's/@@KERNEL_IMAGETYPE@@/${KERNEL_IMAGETYPE}/' \
         -e 's/@@KERNEL_DTB_PREFIX@@/${DTB_PREFIX}/' \
         -e 's/@@FITCONF_FDT_OVERLAYS@@/${FITCONF_FDT_OVERLAYS}/' \
+        -e 's/@@BANK_WORD@@/${BANK_WORD}/ ' \
+        -e 's/@@FUSE_NUM@@/${FUSE_NUM}/' \
+        -e 's/@@STATUS_CMD@@/${STATUS_CMD}/' \
+        -e 's/@@CLOSE_CMD@@/${CLOSE_CMD}/' \
+        -e "s/@@VARS_TO_DEL@@/${vars_to_del}/" \
         ${WORKDIR}/uEnv.txt.in > uEnv.txt
```

Comment:
Done in V3, though I had to slightly modify your suggestion because the parser Yocto runs on each recipe file really did not like that expression you gave.

For one the parser seems to error on any kind of arithmetic expansion. Even just a simple `foo=$((1+1)` gave a parsing error. I got around this by just using `let` for the arithmetic operation and that seems to not make the parser angry.

Second, the new format of the `BANK_WORD` input also made the parser unhappy. But only when it was defined outside of `do_compile`. If I defined it inside `do_compile` then it didn't complain for some reason even though the values should be equivalent(?). To get around this `BANK_WORD` is still initially defined outside of `do_compile`, but then I reassign it to another variable inside `do_compile` and use that for the loop. For whatever reason this keeps the parser happy. My only guess is that there is some invisible/minuscule difference between the variable contents when defined outside of shell functions like `do_compile` versus inside these functions.

Reply:
Glad you managed to deal with the Bitbake idiosyncrasies :smile_cat:.